arrow/src/datatypes/datatype.rs

```diff
@@ -479,57 +1015,38 @@ pub(crate) fn validate_decimal_precision(value: i128, precision: usize) -> Resul
     }
 }
 
-/// Validates that the specified string value can be properly
-/// interpreted as a Decimal256 number with precision `precision`
+/// Validates that the specified `byte_array` of little-endian format
+/// value can be properly interpreted as a Decimal256 number with precision `precision`
 #[inline]
-pub(crate) fn validate_decimal256_precision(
-    value: &BigInt,
+pub(crate) fn validate_decimal256_precision_with_lt_bytes(
+    lt_value: &[u8],
     precision: usize,
-) -> Result<&BigInt> {
+) -> Result<()> {
     if precision > DECIMAL256_MAX_PRECISION {
         return Err(ArrowError::InvalidArgumentError(format!(
-            "Max precision of a Decimal256 is {}, but got {}",
+            "Max precision of a Decima256 is {}, but got {}",
```

Comment:
Original is correct, why change it?

```suggestion
            "Max precision of a Decimal256 is {}, but got {}",
```